scripts/lean-pr-testing-comments.sh

```diff
@@ -0,0 +1,105 @@
+      -H "Authorization: Bearer $TOKEN" \
+      -H "X-GitHub-Api-Version: 2022-11-28" \
+      https://api.github.com/repos/leanprover/lean4/issues/$pr_number/labels/breaks-mathlib
+    echo "Adding label builds-mathlib"
+    curl -L -s \
+      -X POST \
+      -H "Accept: application/vnd.github+json" \
+      -H "Authorization: Bearer $TOKEN" \
+      -H "X-GitHub-Api-Version: 2022-11-28" \
+      https://api.github.com/repos/leanprover/lean4/issues/$pr_number/labels \
+      -d '{"labels":["builds-mathlib"]}'
+  else
+    echo "Removing label builds-mathlib"
+    curl -L -s \
+      -X DELETE \
+      -H "Accept: application/vnd.github+json" \
+      -H "Authorization: Bearer $TOKEN" \
+      -H "X-GitHub-Api-Version: 2022-11-28" \
+      https://api.github.com/repos/leanprover/lean4/issues/$pr_number/labels/builds-mathlib
+    echo "Adding label breaks-mathlib"
+    curl -L -s \
+      -X POST \
+      -H "Accept: application/vnd.github+json" \
+      -H "Authorization: Bearer $TOKEN" \
+      -H "X-GitHub-Api-Version: 2022-11-28" \
+      https://api.github.com/repos/leanprover/lean4/issues/$pr_number/labels \
+      -d '{"labels":["breaks-mathlib"]}'
+  fi
+
+  # Use GitHub API to check if a comment already exists
+  existing_comment=$(curl -L -s -H "Authorization: token $TOKEN" \
+                          -H "Accept: application/vnd.github.v3+json" \
+                          "https://api.github.com/repos/leanprover/lean4/issues/$pr_number/comments" \
+                          | jq '.[] | select(.body | startswith("✅ Mathlib") or startswith("❌ Mathlib") or startswith("💥 Mathlib"))')
+  existing_comment_id=$(echo "$existing_comment" | jq -r .id)
+  existing_comment_body=$(echo "$existing_comment" | jq -r .body)
+
+  branch="[lean-pr-testing-$pr_number](https://github.com/leanprover-community/mathlib4/compare/master...lean-pr-testing-$pr_number)"
+  # Depending on the success/failure, set the appropriate message
+  if [ "$LINT_OUTCOME" == "success" ]; then
+    message="✅ Mathlib branch $branch has successfully built against this PR. ($current_time)"
+  elif [ "$TEST_OUTCOME" == "success" ]; then
+    message="❌ Mathlib branch $branch built against this PR, but linting failed. ($current_time) [View Log]($WORKFLOW_URL)"
+  elif [ "$BUILD_OUTCOME" == "success" ]; then
+    message="❌ Mathlib branch $branch built against this PR, but testing failed. ($current_time) [View Log]($WORKFLOW_URL)"
+  else
+    message="💥 Mathlib branch $branch failed against this PR. ($current_time) [View Log]($WORKFLOW_URL)"
+  fi
+
+  echo $message
+
+  # Append new result to the existing comment or post a new comment
+  if [ -z "$existing_comment_id" ]; then
+    # Post new comment with a bullet point
+    echo "Posting as new comment at leanprover/lean4/issues/$pr_number/comments"
+    curl -L -s \
+      -X POST \
+      -H "Authorization: token $TOKEN" \
+      -H "Accept: application/vnd.github.v3+json" \
+      -d "{\"body\":\"- $message\"}" \
```

Comment:
to make the escaping pain go away:
```suggestion
      -d "$(jq --null-input --arg val "$message" '{"message": $val}')"
```

Reply:
(and below)

Reply:
That seems more painful to me? And the `- ` got dropped.

Reply:
(the `-` isn't dropped as of my edit, which was 40 minutes before your comment; perhaps you had a stale version of the page?)

Reply:
`message` is defined a few lines above, and doesn't have any `"`s.

Reply:
But sure. Sorry, I am pretty tired of testing this script with the hour long wait for a live test, and so feeling very cautious about changes that haven't already been tested.

Reply:
In the case where this is updating an existing message, a `"` could slip in.